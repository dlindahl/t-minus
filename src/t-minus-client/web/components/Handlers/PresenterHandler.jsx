import { bindActionCreators } from 'redux'
import { Component, PropTypes } from 'react'
import { connect } from 'react-redux'
import PresenterLayout from '../Layout/PresenterLayout'
import { presenterReady } from '../../../shared/actions/PresenterActions'
import Teleprompter from '../Teleprompter/Teleprompter'
import TimeDisplay from '../TimeDisplay/TimeDisplay'

function getState (state) {
  return {
    display: state.display,
    percentComplete: state.clock.percentComplete
  }
}

function getActions (dispatch) {
  return bindActionCreators({ presenterReady }, dispatch)
}

@connect(getState, getActions)
export default class PresenterHandler extends Component {
  componentDidMount () {
    this.props.presenterReady(window)
  }
  render () {
    return (
      <PresenterLayout percentComplete={this.props.percentComplete}>
        <TimeDisplay {...this.props.display}/>
        <Teleprompter text={this.props.display.teleprompter}/>
      </PresenterLayout>
    )
  }
}

PresenterHandler.propTypes = {
  display: PropTypes.object,
  percentComplete: PropTypes.number,
  presenterReady: PropTypes.func
}
