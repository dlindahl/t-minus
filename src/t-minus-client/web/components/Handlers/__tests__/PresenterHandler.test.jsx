import configureStore from 'redux-mock-store';
import emptyObj from 'empty/object';
import PresenterHandler from '../PresenterHandler';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

const STATES = {
  default: {
    clock: {
      percentComplete: 0
    },
    display: {
      teleprompter: 'TEXT'
    },
    meta: emptyObj
  }
};
const store = configureStore();

describe('PresenterHandler', () => {
  it('renders the component', () => {
    // Mute the console
    const unmockedWarn = console.warn;
    console.warn = jest.fn();
    const component = renderer.create(
      <Provider store={store(STATES.default)}>
        <PresenterHandler/>
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(console.warn).toHaveBeenCalled();
    console.warn = unmockedWarn;
  });

  it('indicates that the Presenter is ready', () => {
    const presenterReady = jest.fn();
    const component = renderer.create(
      <Provider store={store(STATES.default)}>
        <PresenterHandler.WrappedComponent
          {...STATES.default}
          presenterReady={presenterReady}
        />
      </Provider>
    );
    expect(presenterReady).toHaveBeenCalledWith(window);
  });
});
