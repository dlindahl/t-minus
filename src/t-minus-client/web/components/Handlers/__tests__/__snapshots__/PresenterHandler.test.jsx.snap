exports[`PresenterHandler renders the component 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }>
  <div
    style={
      Object {
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "height": "100vh",
      }
    }>
    <main
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
        }
      }>
      <div
        style={
          Object {
            "alignItems": "center",
            "background": "var(--Clouds)",
            "color": "var(--MidnightBlue)",
            "display": "flex",
            "flex": 1,
            "flexDirection": "column",
            "fontFamily": "Roboto Mono, monospace",
            "fontSize": "14vw",
            "justifyContent": "center",
            "width": "100%",
          }
        }>
        <div>
          <span
            data-test-id="prefix"
            style={
              Object {
                "opacity": 0,
                "pointerEvents": "none",
              }
            }>
            -
          </span>
          <span
            data-test-id="hours">
            --
          </span>
          :
          <span
            data-test-id="minutes">
            --
          </span>
          :
          <span
            data-test-id="seconds">
            --
          </span>
          <span
            data-test-id="microseconds"
            style={
              Object {
                "color": "var(--Silver)",
                "fontSize": "5vw",
              }
            }>
            .
            ----
          </span>
        </div>
      </div>
      <div
        style={
          Object {
            "alignItems": "center",
            "background": "var(--Clouds)",
            "color": "var(--MidnightBlue)",
            "display": "flex",
            "flex": 1,
            "fontFamily": "Roboto Mono, monospace",
            "fontSize": "8vw",
            "justifyContent": "center",
            "overflow": "hidden",
            "paddingBottom": 10,
            "paddingLeft": "5%",
            "paddingRight": "5%",
            "paddingTop": 10,
            "textTransform": "uppercase",
            "transition": "flex-grow .5s ease-in-out",
          }
        }>
        <div
          data-test-id="prompterText">
          TEXT
        </div>
      </div>
    </main>
    <div
      style={
        Object {
          "background": "var(--Clouds)",
        }
      }>
      <div
        style={
          Object {
            "background": "var(--Clouds)",
            "height": 50,
            "whiteSpace": "nowrap",
            "width": "0%",
          }
        } />
    </div>
  </div>
</div>
`;
